searxng/searxng
# Simple theme: allow the user to choose auto, light, dark style

## What does this PR do?
Simple theme: this PR allows the user choose the theme style:
* auto: it follow the browser preferences (same as now)
* light: force the light style
* dark: force the dark style

![image](https://user-images.githubusercontent.com/1594191/142638749-70fd6cec-eefc-4c6c-8c25-51ecf644d6b8.png)

----
Implementation:
* in definition.less, the values of the dark style are defined in a LESS mixins, which to define them twice (once for `:root.theme-dark`, once for the media query).
* `settings.yml`: `ui.theme_args.simple_style` (underscore)
* new cookie and value in preferences.py: `simple-style` (dash like `oscar-style`)
* the choose auto, light, dark can be translated
* the html element contains the user choice. For example : `<html class="no-js theme-dark" lang="fr-FR" >`. 
```
<html class="no-js theme-dark" lang="en-EN" >
```
* `searx/static/themes/simple/src/js/head/00_init.js` is changed not to erase this new CSS class.
* the CSS about the `select` element also make a detection about style.

## Why is this change important?

One step further to change the default theme.

## How to test this PR locally?

matrix tests:
* in the SearXNG preferences: light, dark, auto
* in the browser preferences: light, dark

Each time:
* check if the SearXNG preferences are followed
* check the icon of the `select` HTML element: 

dark: ![image](https://user-images.githubusercontent.com/1594191/142640817-46cdc104-5125-479a-b0be-8233625034d4.png)
light: ![image](https://user-images.githubusercontent.com/1594191/142640949-56e1ffa6-0121-44f3-b86a-b623f1d50f12.png)


## Author's checklist

* The new cookie is `simple-style`, may be it should be `simple_style` ? (the only exception to use a dash is `oscar-style`).
* In a second step, it is possible to let the user change the style directly from the `/` and `/search`

## Related issues

Related to https://github.com/searxng/searxng/issues/463#issuecomment-955787945
